### PR TITLE
Add Plasma Forge (Endgame multi)

### PR DESCRIPTION
**Pull request by Connor-Colenso**
Adds plasma forge multi. Endgame smelter multiblock. Still need to add recipes for the multi itself but I'll do that inside TT. Have added a few recipes for the multi but these will likely need adjusting and feedback from others. Probably made a lot of random mistakes code wise as I've not done this before so could use a good look through. Tested and loads in the full pack, multi works.

Screenshot 2022-06-08 at 18 42 31

### CODE REVIEW

**Comment by Steelux8** [comment]
Will this use noble gasses at all? I'm working on rebalancing their usage on EBFs, knowing that would be good.

**Comment by Eranicus** [comment]
Looks absolutely awesome, just make sure it is crazy expensive to make :)

**Comment by Pxx500** [comment]
Is there some kind of main chamber? Where does the smelting happen?

**Comment by Connor-Colenso** [comment]
> Is there some kind of main chamber? Where does the smelting happen?

If you really want some lore larp then the smelting is done by moving items through a parallel spacetime containing abundant energy which can transform and heat it up. Something, something technobabble.

**Comment by Pxx500** [comment]
> > Is there some kind of main chamber? Where does the smelting happen?
> 
> If you really want some lore larp then the smelting is done by moving items through a parallel spacetime containing abundant energy which can transform and heat it up. Something, something technobabble.

image

**Comment by Connor-Colenso** [comment]
Many to many fusion support implemented for catalyst recipes.

image

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PlasmaForge.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+                    {"                                 ", "         N   N     N   N         ", "         N   N     N   N         ", "         N   N     N   N         ", "                                 ", "                                 ", "                                 ", "         N   N     N   N         ", "         N   N     N   N         ", " NNN   NNN   N     N   NNN   NNN ", "                                 ", "                                 ", "                                 ", " NNN   NNN             NNN   NNN ", "                                 ", "                                 ", "                                 ", "                                 ", "                                 ", " NNN   NNN             NNN   NNN ", "                                 ", "                                 ", "                                 ", " NNN   NNN             NNN   NNN ",}
```
Comment (GlodBlock) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PlasmaForge.java:94]: it is crossed chuck now, you should force load all chucks when it is running.

**Comment by Connor-Colenso** [comment]
Chunkloading seems to work (ran away for a while and it kept producing stuff) so I'm marking that as working.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PlasmaForge.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+                    {"         N   N     N   N         ", "         bCCCb     bCCCb         ", "  N      bCCCb     bCCCb      N  ", "         bCCCb     bCCCb         ", "         NCCCN     NCCCN         ", "         NCCCN     NCCCN         ", "         NCCCN     NCCCN         ", "         bCCCb     bCCCb         ", "         bCCCb     bCCCb         ", "NbbbNNNbbNCCCb     bCCCNbbNNNbbbN", " CCCCCCCCC   N     N   CCCCCCCCC ", " CCCCCCCCC             CCCCCCCCC ", " CCCCCCCCC             CCCCCCCCC ", "NbbbNNNbbbN           NbbbNNNbbbN", "  N     N               N     N  ", "  N     N               N     N  ", "                                 ", "  N     N               N     N  ", "  N     N               N     N  ", "NbbbNNNbbbN           NbbbNNNbbbN", " CCCCCCCCC             CCCCCCCCC ", " CCCCCCCCC             CCCCCCCCC ", " CCCCCCCCC             CCCCCCCCC ", "NbbbNNNbbbN    NbN    NbbbNNNbbbN",},
```
Comment (GlodBlock) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PlasmaForge.java:78]: these diagonal connected structures need to be avoided. gt only checks 6 faces connected block when placing block. you can build the multi and then remove these diagonal connected blocks

Comment (Connor-Colenso) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_PlasmaForge.java:78]: This is really hard to abuse in-game due to stuff like reloading the world resetting it so as discuss on discord I think we can ignore it here but it is certainly something to keep in mind for future.

File: src/main/java/gregtech/api/util/GT_Recipe.java [hunk 3]
```diff
@@ -769,7 +769,7 @@ public static class GT_Recipe_Map {
         public static final GT_Recipe_Map sPrinterRecipes = new GT_Recipe_Map_Printer(new HashSet<>(5), "gt.recipe.printer", "Printer", null, RES_PATH_GUI + "basicmachines/Printer", 1, 1, 1, 1, 1, E, 1, E, true, true);
         public static final GT_Recipe_Map sSifterRecipes = new GT_Recipe_Map(new HashSet<>(105), "gt.recipe.sifter", "Sifter", null, RES_PATH_GUI + "basicmachines/Sifter", 1, 9, 0, 0, 1, E, 1, E, true, true);
         public static final GT_Recipe_Map sPressRecipes = new GT_Recipe_Map_FormingPress(new HashSet<>(300), "gt.recipe.press", "Forming Press", null, RES_PATH_GUI + "basicmachines/Press", 2, 1, 2, 0, 1, E, 1, E, true, true);
-        public static final GT_Recipe_Map sLaserEngraverRecipes = new GT_Recipe_Map(new HashSet<>(810), "gt.recipe.laserengraver", "Precision Laser Engraver", null, RES_PATH_GUI + "basicmachines/LaserEngraver", 2, 1, 2, 0, 1, E, 1, E, true, true);
+        public static final GT_Recipe_Map sLaserEngraverRecipes = new GT_Recipe_Map(new HashSet<>(810), "gt.recipe.laserengraver", "Precision Laser Engraver", null, RES_PATH_GUI + "basicmachines/LaserEngraverNEI", 2, 1, 0, 0, 1, E, 1, E, true, true);
```
Comment (GlodBlock) [inline src/main/java/gregtech/api/util/GT_Recipe.java:772]: you only change the recipe map and the laser engraver still won't aceept fluid. you need to give fluid tank cap for singleblock laser engraver.

Comment (Connor-Colenso) [reply src/main/java/gregtech/api/util/GT_Recipe.java:772]: Multiblock is intended purpose of these recipes but I've aded 8kL per tier just incase someone decides to add recipes for the singleblock.

**Review by boubou19 — APPROVED** [review]

**Pull request merged**